client/src/includes/panels.ts

```diff
@@ -69,13 +73,67 @@ function initCollapsiblePanel(toggle: HTMLButtonElement) {
   content.addEventListener('beforematch', togglePanel.bind(null, true));
 }
 
+/**
+ * Initialises event handlers for collapsing / expanding all panels
+ */
+const initAllCollapsiblePanels = (toggles) => {
+  let collapseAllToggle = document.querySelector<HTMLButtonElement>(
```

Comment:
It would be much nicer if we could instead add event listeners to each panel to trigger the collapse/ expand behaviour. 

This way other customisations could be built more easily. 

Similar to how the new dialog component handles this kind of external close behaviour. Although we should use a more appropriate event name like `wagtail:panel-collapse`

Reply:
Can you point me to the dialog functionality you're referring to @lb- ? I'm not quite sure I understand what you mean.

Reply:
@lb- could you expand on why you’d think it’s much nicer? I can kind of see it in some ways, but we’d still have to trigger the event on all the panels as I understand it, so it’s not clear to me it’s that much of a win.

Note if we introduce events like this we also need to make sure they’re properly registered and de-registered based on what happens on the page, so to me it’s not that clear-cut of a win, at least for components that are within our control anyway.

Reply:
Update on my point above regarding when the user collapses all sections individually - I discussed this with both @thibaudcolas and @benenright - it would mean making the code a bit more complex and slightly less performant - we'd need to check wether all panels are closed on each open / close of an individual panel. We're leaving it for now but it could be a future enhancement.